File: ERP_Anass_app/src/app/models/Inventory/StockMovement.ts

export enum StockMovementType {
    Purchase = 0,
    Sale = 1,
    Adjustment = 2,
    Transfer = 3
}

export interface StockMovement {
    idStockMovement?: number;
    warehouseID?: number;
    type?: StockMovementType;
    movementDate?: Date;
    warehouse?: any;
    stockMovementDetails?: StockMovementDetail[];
}

export interface StockMovementDetail {
    idStockMovementDetail?: number;
    stockMovementId?: number;
    articleID?: number;
    quantity?: number;
    article?: any;
}

File: ERP_SERVER/ERP_Anass_backend/Models/StockMovement.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace ERP_Anass_backend.Models
{
    public enum StockMovementType
    {
        Purchase,
        Sale,
        Adjustment,
        Transfer
    }

    public class StockMovement
    {
        [Key]
        public int idStockMovement { get; set; }
        // Removed ArticleID and Quantity from Header
        public int WarehouseID { get; set; }
        public StockMovementType Type { get; set; }
        public DateTime MovementDate { get; set; } = DateTime.UtcNow;

        [JsonIgnore]
        public Warehouse? Warehouse { get; set; }
        
        public ICollection<StockMovementDetails>? StockMovementDetails { get; set; }
    }
}
